full-stack-tests/imodelhub-client/src/imodelhub/iModels.test.ts

```diff
@@ -858,15 +852,21 @@ describe("iModelHub iModelsHandler", () => {
   });
 
   it("should handle special characters in get by name query", async () => {
-    const name = `Д - ${Guid.createValue()}`;
-    const description = "Test iModel created by imodeljs-clients tests";
-    mockCreateEmptyiModel(projectId, Guid.createValue(), name, description);
-    await imodelClient.iModels.create(requestContext, projectId, name, { description });
+    mockGetIModelByName(projectId, imodelName);
+    let imodel: HubIModel = (await iModelClient.iModels.get(requestContext, projectId, new IModelQuery().byName(imodelName)))[0];
+    const oldimodelName = imodel.name;
 
-    mockGetIModelByName(projectId, name);
-    const iModels = await imodelClient.iModels.get(requestContext, projectId, new IModelQuery().byName(name));
+    const newName = `Д - ${Guid.createValue()}`;
+    imodel.name = newName;
+    mockUpdateiModel(projectId, imodel);
+    imodel = await iModelClient.iModels.update(requestContext, projectId, imodel);
```

Comment:
If this test fails at any point after this line, iModel will no longer be deleted.

Reply:
Moved the name declaration up to the test suite scope and called the delete in after()